src/providers/ObsEditProvider.js

```diff
@@ -105,7 +105,8 @@ const ObsEditProvider = ( { children }: Props ): Node => {
     try {
       const realm = await Realm.open( realmConfig );
       const obs = realm.objectForPrimaryKey( "Observation", savedUUID );
-      setObservations( [obs] );
+      const plainObject = Observation.mapPlainObjectForObsEdit( obs );
```

Comment:
Yes, that's the problem I was running into. Without converting to POJOs, there's no way to update an attribute without using a write transaction to save to Realm. We don't want to modify the Realm observation until a user actually taps "save" or "upload," so we need a way to track attribute changes in the meantime, while also displaying the attribute changes in ObsEdit.

Reply:
Hm, clearly [we're not the first to encounter this](https://github.com/realm/realm-js/issues/2518). I would see if there's a way we can use a version of Realm that supports that `toJSON` method, otherwise we should implement one of the more generic solutions in that threat so we don't have to re-implement these `toPlainObject` methods for every model.